src/esp/assets/MeshMetaData.h

```diff
@@ -5,10 +5,46 @@
 #pragma once
 
 #include "esp/core/esp.h"
+#include "esp/gfx/magnum.h"
 
 namespace esp {
 namespace assets {
 
+//! Some mesh files include a transformation hierarchy. A @ref MeshTransformNode
+//! stores this hierarchy and indices for the meshes at each level such that it
+//! can be reused to instances meshes later.
+struct MeshTransformNode {
+  //! mesh ID within @ref MeshMetaData::meshIndex
+  int meshIDLocal;
+
+  //! material ID within @ref MeshMetaData::materialIndex
+  int materialIDLocal;
+
+  //! Object ID in the original file
+  int componentID;
+
+  std::vector<MeshTransformNode> children;
+
+  //! Node local transform
+  Magnum::Matrix4 T;
+
+  MeshTransformNode() {
```

Comment:
Why is this constructor needed? If you omit it, you can't create an undefined object and trying to create such an object will error at compile time.